include/drivers/rtc.h

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2020 Paratronic
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_RTC_H_
+#define ZEPHYR_INCLUDE_DRIVERS_RTC_H_
+
+/**
+ * @file
+ * @brief Public RTC driver APIs
+ */
+
+#include <zephyr/types.h>
+#include <time.h>
+#include <device.h>
+
+typedef int (*rtc_api_gettime_t)(struct device *dev, struct timespec *tp);
+typedef int (*rtc_api_settime_t)(struct device *dev, const struct timespec *tp);
+
+struct rtc_driver_api {
+	rtc_api_gettime_t gettime;
+	rtc_api_settime_t settime;
+};
+
+/**
+ * @brief Function to get RTC time
+ *
+ * @param[in] dev RTC device
+ * @param[out] tp pointer to where to store the current rtc value
+ *
+ * @return 0 on success, negative on error
+ */
+__syscall int rtc_gettime(struct device *dev, struct timespec *tp);
```

Comment:
I forgot to mention: could you provide the drivers/rtc/rtc_handlers.c in a separate patch for userland support?

Reply:
I did it!